test/unittest/Database/DatabaseLoadTests.cpp

```diff
@@ -0,0 +1,1108 @@
+    check_is_double(dump, dump[LOCATOR_CONTAINER_TAG].begin().value()[DATA_VALUE_DATA_TAG]
+            [DATA_KIND_NETWORK_LATENCY_TAG].begin().value().begin().value()[DATA_VALUE_DATA_TAG]);
+
+}
+
+
+class DataBaseTest : public Database
+{
+public:
+
+    const std::map<EntityId, std::shared_ptr<Domain>>& domains()
+    {
+        return domains_;
+    }
+
+};
+
+// Test the load of a database with wrong references
+TEST(database_load_tests, load_wrong_references)
+{
+    // Read JSON
+    DatabaseDump dump = load_file(SIMPLE_DUMP_FILE);
+
+    // Host reference user that no exists
+    {
+        DatabaseDump dumpCopy = dump;
+
+        DatabaseDump container = DatabaseDump::object();
+
+        DatabaseDump entity_info = DatabaseDump::object();
+        entity_info[NAME_INFO_TAG] = "host_1";
+        entity_info[USER_CONTAINER_TAG] = DatabaseDump::array({"200"});
+        container["100"] = entity_info;
+
+        dumpCopy[HOST_CONTAINER_TAG].insert(container.begin(), container.end());
+
+        Database db;
+        ASSERT_THROW(db.load_database(dumpCopy), CorruptedFile);
+    }
+
+    // User reference host that no exists
+    {
+        DatabaseDump dumpCopy = dump;
+
+        DatabaseDump container = DatabaseDump::object();
+
+        DatabaseDump entity_info = DatabaseDump::object();
+        entity_info[NAME_INFO_TAG] = "user_1";
+        entity_info[HOST_ENTITY_TAG] = "200";
+        entity_info[PROCESS_CONTAINER_TAG] = DatabaseDump::array();
+        container["100"] = entity_info;
+
+        dumpCopy[USER_CONTAINER_TAG].insert(container.begin(), container.end());
+
+        Database db;
+        ASSERT_THROW(db.load_database(dumpCopy), CorruptedFile);
+    }
+
+    // Host reference user, which one is not referencing the host
+    {
+        DatabaseDump dumpCopy = dump;
+
+        DatabaseDump container = DatabaseDump::object();
+
+        DatabaseDump entity_info = DatabaseDump::object();
+        entity_info[NAME_INFO_TAG] = "host_1";
+        entity_info[USER_CONTAINER_TAG] = dumpCopy[USER_CONTAINER_TAG].begin().key();
+        container["100"] = entity_info;
+
+        dumpCopy[HOST_CONTAINER_TAG].insert(container.begin(), container.end());
+
+        Database db;
+        ASSERT_THROW(db.load_database(dumpCopy), CorruptedFile);
+    }
+
+    // User reference host, which one is not referencing the user
+    {
+        DatabaseDump dumpCopy = dump;
+
+        DatabaseDump container = DatabaseDump::object();
+
+        DatabaseDump entity_info = DatabaseDump::object();
+        entity_info[NAME_INFO_TAG] = "user_1";
+        entity_info[HOST_ENTITY_TAG] = dumpCopy[HOST_CONTAINER_TAG].begin().key();
+        entity_info[PROCESS_CONTAINER_TAG] = DatabaseDump::array();
+        container["100"] = entity_info;
+
+        dumpCopy[USER_CONTAINER_TAG].insert(container.begin(), container.end());
+
+        Database db;
+        ASSERT_THROW(db.load_database(dumpCopy), CorruptedFile);
+    }
+
```

Comment:
I am missing a lot of other test cases in this specific test. Please complete the test taking into account all possible entities.